app/api/include/api/sonicpi_api.h

```diff
@@ -0,0 +1,382 @@
+
+struct CueInfo
+{
+    std::string time;
+    std::string address;
+    int id;
+    std::string args;
+    uint64_t index;
+    TimePoint arrivalTime;
+};
+
+// This is the processed audio data from the thread
+struct ProcessedAudio
+{
+    std::vector<float> m_spectrum[2];
+    std::vector<float> m_spectrumQuantized[2];
+    std::vector<float> m_samples[2];
+    std::vector<float> m_monoSamples;
+};
+
+enum class MessageType
+{
+    StartupError,
+    RuntimeError,
+    SyntaxError,
+    Message,
+    Info,
+    InfoText,
+    Muti
+};
+
+struct MessageData
+{
+    std::string text;
+    int style = 0;
+};
+
+struct MessageInfo : MessageData
+{
+    MessageType type;
+    int jobId = 0;
+    std::string threadName;
+    std::string runtime;
+    std::string backtrace;
+    int line = 0;
+    std::string errorLineString;
+    std::string lineNumString;
+
+    std::vector<MessageData> multi;
+};
+
+enum class MidiType
+{
+    Out,
+    In
+};
+
+struct MidiInfo
+{
+    MidiType type;
+    std::string portInfo;
+};
+
+enum class StatusType
+{
+    Ack,
+    AllComplete,
+    Exited
+};
+
+struct StatusInfo
+{
+    StatusType type;
+    std::string id;
+};
+
+struct VersionInfo
+{
+    std::string version;
+    int num;
+    std::string latestVersion;
+    int latestVersionNum;
+    int lastCheckedDay;
+    int lastCheckedMonth;
+    int lastCheckedYear;
+    std::string platform;
+};
+
+enum class BufferType
+{
+    Replace,
+    ReplaceIndex,
+    ReplaceLines,
+    RunIndex
+};
+
+struct BufferInfo
+{
+    BufferType type;
+    std::string id;
+    int bufferIndex;
+    std::string content;
+    int line;
+    int index;
+    int lineNumber;
+
+    // replace-lines
+    int startLine;
+    int finishLine;
+    int pointLine;
+    int pointIndex;
+};
+
+// Callbacks from the API into your application/UI.
+// These are not on the UI thread, and marshalling is left up to the client
+struct IAPIClient
+{
+    virtual void Report(const MessageInfo& info) = 0;
+    virtual void Status(const StatusInfo& info) = 0;
+    virtual void Cue(const CueInfo& info) = 0;
+    virtual void Midi(const MidiInfo& info) = 0;
+    virtual void Version(const VersionInfo& info) = 0;
+    virtual void AudioDataAvailable(const ProcessedAudio& audio) = 0;
+    virtual void Buffer(const BufferInfo& info) = 0;
+};
+
+// Always UDP
+enum class APIProtocol
+{
+    UDP = 0,
+    TCP = 1
+};
+
+struct APISettings
+{
+    bool logSynths = true;
+    bool logCues = false;
+    bool checkArgs = true;
+    bool enableExternalSynths = true;
+    bool timingGuarantees = false;
+    int defaultMidiChannel = -1;
+
+    void Preprocess(std::string& code) const
+    {
+        if (!logSynths)
+        {
+            code = "use_debug false #__nosave__ set by Qt GUI user preferences.\n" + code;
+        }
+
+        if (!logCues)
+        {
+            code = "use_cue_logging false #__nosave__ set by Qt GUI user preferences.\n" + code;
+        }
+
+        if (!checkArgs)
```

Comment:
This condition is reversed.
```suggestion
        if (checkArgs)
```